File: pf-core/src/main/java/com/purdynet/persistence/MongoUtils.java
package com.purdynet.persistence;

import com.mongodb.MongoClient;

/**
 * Created with IntelliJ IDEA.
 * User: dnpurdy
 * Date: 9/16/13
 * Time: 6:50 PM
 * To change this template use File | Settings | File Templates.
 */
public class MongoUtils
{
    private static final String host = "localhost";

    public static MongoClient getClient()
    {
        try {
            return new MongoClient( host );
        }
        catch (Exception e)
        {
            throw new RuntimeException("Failed to connect to mongo!", e);
        }
    }
}

File: pf-core/src/main/scala/com/purdynet/persistence/MongoUtils.scala
package com.purdynet.persistence

import com.mongodb.MongoClient

/**
 * Created with IntelliJ IDEA.
 * User: dnpurdy
 * Date: 11/17/13
 * Time: 8:42 PM
 */
object MongoUtils {
  def getClient: MongoClient = {
    getClient("localhost")
  }

  def getClient(host:String): MongoClient = {
    try
      new MongoClient(host)

    catch {
      case e: Exception => {
        throw new RuntimeException("Failed to connect to mongo!", e)
      }
    }
  }
}
